include/seqan3/core/detail/pack_algorithm.hpp

```diff
@@ -0,0 +1,220 @@
+ * function template declaration. To avoid adding an additional impl function for every algorithm, which takes care of
+ * the unpacking of the parameters, this helper struct provides an invocation method to pass the wrapped parameters
+ * to the desired parameter pack version of the algorithm. Since the target function is iteself a function template
+ * we cannot simply pass a pointer to that function without knowing the types of it's argument before. Accordingly,
+ * one needs to defer the invocation in a lambda function which is called in the static
+ * seqan3::detail::type_identity_delegate::invoke method.
+ */
+template <template <typename ...> typename list_t, typename ...args_t>
+struct type_identity_delegate<list_t<args_t...>>
+{
+    /*!\brief Invokes the actual function by passing the types as instances of std::type_identity to the target
+     *        function.
+     * \tparam wrapper_t The type of the function wrapper.
+     * \param[in] wrapper The wrapper function, e.g. a lambda function.
+     * \returns The std::invoke_result of calling wrapper with the parameters.
+     *
+     * \details
+     *
+     * Invokes the wrapper by passing the types wrapped in std::type_identity.
+     */
+    template <typename wrapper_t>
+    //!\cond
+        requires std::invocable<wrapper_t, std::type_identity<args_t>...>
+    //!\endcond
+    static constexpr std::invoke_result_t<wrapper_t, std::type_identity<args_t>...> invoke(wrapper_t && wrapper)
+    {
+        return wrapper(std::type_identity<args_t>{}...);
+    }
+};
+
+//-----------------------------------------------------------------------------
+// all_of
+//-----------------------------------------------------------------------------
+
+/*!\brief Tests whether a given predicate evaluates to `true` for each element in a parameter pack.
+ * \ingroup core
+ *
+ * \tparam unary_predicate_t The function type, like function pointers, functors and lambdas;
+ *                           must model std::predicate expanded on each argument type.
+ * \tparam pack_t The parameter pack of the arguments (each argument type can be different).
+ *
+ * \param[in] fn The predicate to evaluate for every argument.
+ * \param[in] args The parameter pack.
+ *
+ * \details
+ *
+ * This function behaves like std::all_of but on parameter packs. The invocation(s) will be done without any loop.
+ *
+ * ### Example
+ *
+ * \include test/snippet/core/detail/all_of_in_pack.cpp
+ *
+ * ### Complexity
+ *
+ * Linear in the number of elements in the pack.
+ *
+ * \attention Opposed to the std::all_of the argument order is changed, such that the first argument is the unary
+ * predicate to invoke on each argument followed by the arguments.
+ *
+ * \sa https://en.cppreference.com/w/cpp/language/parameter_pack
+ */
+template <typename unary_predicate_t, typename ...pack_t>
+[[nodiscard]] constexpr bool all_of(unary_predicate_t && fn, pack_t && ...args)
```

Comment:
Well, there are many opinions out there that actually are in favor of nodiscard being the default. 
Also from the proposal: 

> You almost never want to discard the return value.  (Using the return value is almost always an essential part of the interface of the given function.) 

I think this pretty much brings it to the point. Not using the return value renders the function useless. So why did the programmer use it then? Did he implied to do something different. Did he not understand the interface? Maybe for_each was the thing he was looking then for. So that is why I am in strong favor to always use nodiscard, except in those interfaces where the return value might give you a "debug" information you do not really need in the program logic. As suggested by the proposal as well.